spec/core/scheduler_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'core/native_scheduler'
+require 'core/scheduler'
+
+describe Core::Scheduler do
+  subject { Core::NativeScheduler.new(poll_interval, heartbeat_interval) }
```

Comment:
We can even probably stub this by creating a stub class that inherits from Core::Scheduler and implements needed methods with stubs. Probably not worth to change in this PR, just thinking out loud.

Reply:
I don't quite understand this. Please do refactor if you think it's necessary.

Reply:
Sounds good, I'll take a look sometime if I can improve this, but the code here is already good enough :)